src/main/java/losty/netatmo/NetatmoHttpClient.java

```diff
@@ -121,41 +154,76 @@ public OAuthJSONAccessTokenResponse login(String email, String password)
 	 * @throws OAuthSystemException
 	 * @throws OAuthProblemException
 	 */
-	public List<Measures> getMeasures(OAuthJSONAccessTokenResponse token, Station station, Module module,
-			List<String> types, String scale, Date dateBegin, Date dateEnd, Integer limit, Boolean realTime)
+	public List<Measures> getMeasures(final OAuthJSONAccessTokenResponse token, final Station station, final Module module,
+			final List<String> types, final String scale, final Date dateBegin, final Date dateEnd, final Integer limit, final Boolean realTime)
 					throws OAuthSystemException, OAuthProblemException {
 
-		String[] typesArr = types.toArray(new String[0]);
+		long dateBeginMillis = 0;
+		if (dateBegin != null) {
+			dateBeginMillis = (dateBegin.getTime() / 1000);
+		}
+		long dateEndMillis = 0;
+		if (dateEnd != null) {
+			dateEndMillis = (dateEnd.getTime() / 1000);
+		}
+
+		return getMeasures(token, station, module, types, scale, dateBeginMillis, dateEndMillis, limit, realTime);
 
-		List<String> params = new ArrayList<>();
+	}
+	
+
+
+	/**
+	 * Returns the list of measures for a device or module owned by the user.
+	 * See
+	 * <a href="https://dev.netatmo.com/en-US/resources/technical/reference/common/getmeasure">
+	 * dev.netatmo.com/en-US/resources/technical/reference/common/getmeasure</a> for more information.
+	 * 
+	 * Some parameters are optional, they can be set to "null".
+	 * 
+	 * @param token The token obtained by the login function.
+	 * @param station The station to query
+	 * @param module The module of the station (optional)
+	 * @param types A list of the types to query
+	 * @param scale The scale to query
+	 * @param dateBegin Start date of the interval to query (optional)
+	 * @param dateEnd End date of the interval to query (optional)
+	 * @param limit The amount of Measures to be returned at maximum (be careful - max. is 1024!)
+	 * @param realTime Some fancy real_time stuff from Netatmo
+	 * @return
+	 * @throws OAuthSystemException
+	 * @throws OAuthProblemException
+	 */
+	public List<Measures> getMeasures(final OAuthJSONAccessTokenResponse token, final Station station, final Module module,
+			final List<String> types, final String scale, final long dateBegin, final long dateEnd, final Integer limit, final Boolean realTime)
+					throws OAuthSystemException, OAuthProblemException {
+
+		final String[] typesArr = types.toArray(new String[0]);
+
+		final List<String> params = new ArrayList<>();
 		params.add("device_id=" + station.getId());
 		params.add("scale=" + scale);
 		params.add("type=" + implode(",", typesArr));
 		if (module != null) {
 			params.add("module_id=" + module.getId());
 		}
-		if (dateBegin != null) {
-			params.add(String.format("date_begin=%d", dateBegin.getTime() / 1000));
+		if (dateBegin > 0) {
```

Comment:
This will prevent all dates before 1st Jan 1970 to be used here. OK - bit theoretically for Netatmo, but still... Do you really need this timestamp-thingy? (...if yes - how about using Long objects?)

Reply:
Good catch - classical copy&paste error ;-) - I think Long objects would also be an option.
I use this time range limitations to batch download the data instead of downloading all of them every time...